docs: update contributing guide with latest instructions (#1529)
nrwl/nx: CONTRIBUTING.md

## Found an Issue?

If you find a bug in the source code or a mistake in the documentation, you can help us by [submitting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-issue) to [our GitHub Repository](https://github.com/nrwl/nx). Even better, you can [submit a Pull Request](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr) with a fix.

## Project Structure

Source code and documentation are included in the top-level folders listed below.

- `docs` - Markdown and configuration files for documentation including tutorials, guides for each supported platform, and API docs.
- `e2e` - E2E tests for Nx packages and schematics.
- `packages` - Source code for Nx packages such as Angular, React, Web, and others including schematics and builders.
- `scripts` - Miscellaneous scripts for project tasks such as building documentation, testing, and code formatting.

## Building the Project

After cloning the project to your machine, to install the dependencies, run:

```bash
yarn
```

To build the `nx`, and `schematics` packages, run:

```bash
yarn build
```

To install the built packages into your `node_modules` folder, run:

```bash

> yarn linknpm

Reviewer: I could be wrong but... I never do this... I don't think it's necessary?

yarn test
```

To test the schematics only, run:

```bash
yarn test:schematics
```

To test the Nx packages only, run:

```bash
yarn test:nx
```

For example, if you need to only run the **ngrx/ngrx.spec.ts** test suite, provide a path to the specific spec file, run:

```bash
yarn test:schematics angular/src/schematics/ngrx/ngrx
```

### Running E2E Tests

To make sure your changes do not break any E2E tests, run:

```bash
yarn e2e
```

> Running E2E tests can take some time, so it is often useful to run a single test. To run tests against only changed files, run:

Reviewer: ```suggestion
Running E2E tests can take some time, so it is often useful to run a single test. To run a single suite of tests, run:
```

- frontend - anything angular specific
- nx - dependency management, basic workspace structure, etc

##### Subject and Body

The subject must contain a description of the change, and the body of the message contains any additional details to provide more context about the change.

Including the issue number that the PR relates to also helps with tracking.

#### Example

```
feat(schematics): add an option to generate lazy-loadable modules

`ng generate lib mylib --lazy` provisions the mylib project in tslint.json

Closes #157
```

#### Commitizen

To simplify and automate the process of committing with this format,
**Nx is a [Commitizen](https://github.com/commitizen/cz-cli) friendly repository**, just do `git add` and execute `yarn commit`.

## Migrations

Nx allows users to automatically upgrade to the newest version of the package. If you are introducing a change that would require the users to upgrade their workspace, add a migration to `packages/schematics/migrations`.

Migrations are named using version numbers in the following fashion: `update-major-minor-patch/update-major-minor-patch.ts`, for example, `update-8-1-0/update-8-1-0.ts`.

> The `yarn nx-migrate` command will run all the migrations after the one encoded in `angular.json`.

Reviewer: We can move this since it is deprecated.
